src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -94,4 +95,214 @@ public void testOffsetAtRead(String cigar, int expectedReferencePos, int posInRe
             SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
             Assert.assertEquals(sam.getReferencePositionAtReadPosition(posInRead), expectedReferencePos);
     }
+
+    @DataProvider(name = "deepCopyTestData")
+    public Object [][] deepCopyTestData() {
+        return new Object[][]{
+                {"3S9M"},
+                {"4M1I6M"},
+        };
+    }
+
+    @Test(dataProvider = "deepCopyTestData")
+    public void testDeepCopyRef(String cigar) {
+        SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
+        testDeepCopy(sam);
+    }
+
+    @Test(dataProvider = "deepCopyTestData")
+    public void testDeepCopyMutate(String cigar) {
+        final SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
+        final byte[] initialBaseQualityCopy = Arrays.copyOf(sam.getBaseQualities(), sam.getBaseQualities().length);
+        final int initialStart = sam.getAlignmentStart();
+
+        final SAMRecord deepCopy = testDeepCopy(sam);
+        Assert.assertTrue(Arrays.equals(sam.getBaseQualities(), deepCopy.getBaseQualities()));
+        Assert.assertTrue(sam.getAlignmentStart() == deepCopy.getAlignmentStart());
+
+        // mutate copy and make sure original remains unchanged
+        final byte[] copyBaseQuals = deepCopy.getBaseQualities();
+        for (int i = 0; i < copyBaseQuals.length; i++) {
+            copyBaseQuals[i]++;
+        }
```

Comment:
Call `setBaseQualities()` here for good measure instead of just modifying the returned array in-place, just in case `getBaseQualities()` is ever changed to perform a defensive copy (as such a change would silently render this test ineffective).